File: 2020-x-sentimental-credit/credit.py
from cs50 import get_string

number = get_string("Insert credit card number:")

def get_digit(number, n):
    return int(number) // 10**n % 10

def get_prve_dve(number):
    return int(number) // 10 ** (len(number) -2) % 100

def get_prvo(number):
    return int(number) // 10 ** (len(number) -1) % 10

def sum_of_sodih(number):
    i = 0
    for g in range(1, len(number), 2):
        i += get_digit(get_digit(int(number), g) * 2, 0) + get_digit(get_digit(int(number), g) * 2, 1)
    #print(f"{i}")
    return i

def sum_of_lihih(number):
    i = 0
    for g in range(0, len(number), 2):
        i += get_digit(int(number), g)
    # print(f"{i}")
    return i

#print(f"{get_prve_dve(number)}")

#print(f"{sum_of_sodih(number) + sum_of_lihih(number)}")

if (sum_of_sodih(number) + sum_of_lihih(number)) % 10 == 0:
    if get_prve_dve(number) == 34 or get_prve_dve(number) == 37:
        print("AMEX")
    elif 50 < get_prve_dve(number) < 56:
        print("MASTERCARD")
    elif get_prvo(number) == 4:
        print("VISA")
else:
    print("INVALID")

File: 2019-x-credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long number = get_long ("Input card number:\n");
    long number1 = number/100000000000000;
    int number2 = number1;
    int sum = 0;
    int sum1 = 0;
    int sum11 = 0;
    int sum22 = 0;
    for (int i = 1; i <17; i++)
    {
        
        if (i % 2 == 0)
        {
            printf("I: %i\n", i);
            sum = number % 10;
            printf("sum:%i\n", sum);
            int sum01 = sum*2;
            printf("sum01:%i\n", sum01);
            if (sum01 > 9)
            {
                sum11 = sum11 + 1 + (sum01 % 10);
            }
            else
            {
                sum11 = sum11 + sum01;
            }
        } 
        else
        {
            sum11 = sum11 + (number % 10);
        }
        number = number /10;
        printf("%i\n", sum11);
    } 
    if (sum11 % 10 != 0)
    {
        printf("INVALID\n");
    }
    else if (number2 == 34 || number2 == 37)
    {
        printf("AMEX\n");
    }
    else if (number2 > 50 && number2 <56)
    {
        printf("MASTERCARD\n");    
    }
       
    else if (number2/10 == 4)
    {
       printf("VISA\n");    
    }
    else
    {
        printf("INVALID\n");
    }
        
}
